rir/src/compiler/translations/rir_2_pir/rir_2_pir.cpp

```diff
@@ -415,7 +417,17 @@ bool Rir2Pir::compileBC(
     case Opcode::Op: {                                                         \
         auto rhs = pop();                                                      \
         auto lhs = pop();                                                      \
-        push(insert(new Name(lhs, rhs, env, srcIdx)));                         \
+        if (typeFeedback[0].numTypes > 0 && typeFeedback[1].numTypes > 0 &&    \
```

Comment:
Wouldn't this work even if we don't have the type feedback? `if (isobj(a) || isobj(b)) then env else elided`

Reply:
Yes, sure it will work. Semantics are preserved. But then whenever binops are used with objects it will trigger a deopt. Deopt should be avoided. Unless we know beforehand that objects are very infrequent?